File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    a1 = [1, 2, 3, 4, 5]
    a2 = [2, 1, 2, 3, 2, 4, 2, 5]
    a3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    x, y, z = 0, 0, 0
    res = [[0, 1], [0, 2], [0, 3]]
    for i in range(len(answers)):
        if a1[x % 5] == answers[i]:
            res[0][0] += 1
        if a2[y % 8] == answers[i]:
            res[1][0] += 1
        if a3[z % 10] == answers[i]:
            res[2][0] += 1
        x += 1
        y += 1
        z += 1
    res.sort(key = lambda x : (-x[0], x[1]))
    answer.append(res[0][1])
    for i in range(1, 3):
        if res[i - 1][0] != res[i][0]:
            break
        answer.append(res[i][1])
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] a = new int[]{1, 2, 3, 4, 5}, b = new int[]{2, 1, 2, 3, 2, 4, 2, 5}, c = new int[]{3,3,1,1,2,2,4,4,5,5};
        int x = 0, y = 0, z = 0;
        for(int i = 0 ; i < answers.length; i++)
        {
            if (a[i % 5] == answers[i]) x++;
            if (b[i % 8] == answers[i]) y++;
            if (c[i % 10] == answers[i]) z++;
        }
        int max = Math.max(x, Math.max(y, z));
        int[] arr = new int[]{x, y, z};
        int len = 0;
        for(int i : arr)
            if (i == max) len++;
        int[] answer = new int[len];
        len = 0;
        for(int i = 0 ; i < 3; i++)
            if (arr[i] == max) answer[len++] = i + 1;
        return answer;
    }
}